src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyCaseBlocksInSwitchStatementSame2.java

```diff
@@ -0,0 +1,119 @@
+/*
+RightCurly
+option = SAME
+tokens = LITERAL_CASE
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyCaseBlocksInSwitchStatementSame2 {
+
+     public static void test9() {
+        int mode = 0;
+        switch (mode) {
+            case 0: int x = 1;
+            case 1: x = 1; break;
+            case 2: {
+                x = 1; break;
+            } default : x = 5;         // violation '}' at column 13 should be alone on a line'
+        }
+    }
+
+    public static void test10() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: int x = 1; break;    // violation '}' at column 13 should be alone on a line'
+            case 2: {
+
+            }
+            default : x = 5;
+        }
+    }
+     public static void test11() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: int x = 1; break;   // violation '}' at column 13 should be alone on a line'
+            case 2: {
+
+            } default : x = 5;            // violation '}' at column 13 should be alone on a line'
+        }
+    }
+    public static void test12() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: {int x = 1;} break;   // violation '}' at column 13 should be alone on a line'
+        }
+    }
+    public static void test13() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            }
+            case 1: {
+
+            }
+            default : break;
+        }
+    }
+     public static void test14() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {
+
+            } case 1: {  }         // violation '}' at column 13 should be alone on a line'
+            default : {break;}
+        }
+    }
+
+    public static void test15() {
+        int mode = 0;
+        switch (mode) {case 0: { } case 1: {  } }
+        // violation above '}' at column 47 should be alone on a line'
```

Comment:
https://checkstyle.org/property_types.html#RightCurlyOption
> same
> Works like alone_or_singleline but .... If no next part of a multi-block statement present, brace must be alone on line. It also allows single-line format of multi-block statements. 

Why isn't this acceptable since it is all on a single line? I see other single lines being allowed, so I am confused.

Reply:
>If no next part of a multi-block statement present, brace must be alone on line

I would not consider this a "multi block statement".

Reply:
I would consider it analogous to this case:
```
➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="RightCurly">
      <property name="option" value="same"/>
    </module>
  </module>
</module>                                                                                                

➜  src cat Test.java 
public class Test {

    public static void main(String[] args) {
        if (args.length > 0) {}  if (args.length > 0) {}
    }

}
                                                                                                              
➜  src javac Test.java 

➜  src java -jar checkstyle-10.13.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:4:31: '}' at column 31 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 1 errors.
```

Reply:
So I assume we are saying this works exactly as `alone_or_singleline` and `}` must always end the line? If so, documentation follows that.

Reply:
> So I assume we are saying this works exactly as `alone_or_singleline` and `}` must always end the line?

Yes but why always? if we have multiple single line blocks format then it is ok to not be at the end of the line 
example from if statements:
```
PS D:\test> cat src/Test3.java
public class Test3 {
  public static void test() {
      int mode = 0;
      if(mode == 0) {} else if(mode == 1) { } else {}
  }
}
PS D:\test> java  -jar checkstyle-10.14.2-all.jar -c config.xml  src/Test3.java
Starting audit...
Audit done.
```